src/symmetric_heap_c.c

```diff
@@ -396,3 +399,27 @@ void SHMEM_FUNCTION_ATTRIBUTES * shmemalign(size_t alignment, size_t size)
 {
     return shmem_align(alignment, size);
 }
+
+
+void SHMEM_FUNCTION_ATTRIBUTES *
+shmemx_malloc_with_hints(size_t size, long hints)
+{
+    void *ret = NULL;
+
+    SHMEM_ERR_CHECK_INITIALIZED();
+
+    if (size == 0) return ret;
+
+    // Check for valid hints
+    if(hints != 0 && hints != SHMEMX_MALLOC_ATOMICS_REMOTE && hints!= SHMEMX_MALLOC_SIGNAL_REMOTE){
```

Comment:
I think `SHMEMX_MALLOC_ATOMICS_REMOTE | SHMEMX_MALLOC_SIGNAL_REMOTE` is also a valid hint. I think it would work to add `#define SHMEMX_MALLOC_MAX_HINT    (1l<<2)` and check `hints >= SHMEMX_MALLOC_MAX_HINT`.

Reply:
Yeah valid point! will fix that. thanks!

Reply:
Is this issue resolved?  I'm not seeing the check against the "maximum" possible hint value.